### PR TITLE
Oauth 2 with doorkeeper

### PR DESCRIPTION
**Pull request by andresgutgon**
## What?
I'm playing with the idea of implement a katuma client in React but also server side rendered. What is called an [Isomorpic App](http://nerds.airbnb.com/isomorphic-javascript-future-web-apps/). In this case this repo with provide only api end points. And the client (in Node) would be responsible of rendering in server/frontend

## Demo in heroku
1. Node Client App: https://katumanode.herokuapp.com
2. This PR is here: https://katumaapi.herokuapp.com/

Login with `frank_example@katuma.org` and the usal `*****2` password :trollface: 

## Node repo
This is the repo for the client app:
https://github.com/andresgutgon/katuma-node

### CODE REVIEW

**Comment by naoisegolden** [comment]
@andresgutgon will this be easy to port to React Native to have a native phone app?

**Comment by andresgutgon** [comment]
> will this be easy to port to React Native to have a native phone app

This PR is not related at all with React or React Native. It's just give Katuma the ability to be an Oauth2 provider. 

IMO Oauth2 is the way to consume an API. So yes, this is helpful for a future Katuma mobile app (whatever technology you want to use: Native, PhoneGap, ReactNative,...)

File: engines/public_pages/app/controllers/public_pages/session_controller.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ def login_attempt
       user = User.find_by(email: params[:email])
       if user && user.authenticate(params[:password])
         session[:current_user_id] = user.id
-        redirect_to '/app/#/dashboard' # TODO put this in some config
+        redirect_after_login_success
```
Comment (enricostano) [inline engines/public_pages/app/controllers/public_pages/session_controller.rb]: May we remove this change since we don't need `SessionController` at all?

File: Gemfile [hunk 2]
```diff
@@ -6,12 +6,24 @@ gem 'oat', '~> 0.4.5'
...
+# Also they recomment manage app start with foreman
+gem 'foreman'
+gem 'puma'
+
 gem 'public_pages', path: "engines/public_pages"
 
+group :assets do
+  gem 'sass-rails'
+  gem 'uglifier'
+end
```
Comment (enricostano) [inline Gemfile]: is this related to this PR or something else?

**Comment by enricostano** [comment]
Many thanks @andresgutgon :clap: :clap: :clap:

File: Gemfile (hunk 2 above)
Comment (andresgutgon) [inline Gemfile]: Deploying to Heroku asset pipeline complained for not have this. Also I removed boostrap sourcemap file
